lib/clickhousex/protocol.ex

```diff
@@ -21,8 +20,7 @@ defmodule Clickhousex.Protocol do
   @ping_query Clickhousex.Query.new("SELECT 1") |> DBConnection.Query.parse([])
   @ping_params DBConnection.Query.encode(@ping_query, [], [])
 
-  @doc false
-  @spec connect(opts :: Keyword.t()) :: {:ok, state} | {:error, Exception.t()}
+  @impl true
```

Comment:
`@impl true` implies `@doc false` and provides better typespec support, at least in vscode :)